proto/dymension/eibc/demand_order.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+package dymensionxyz.dymension.eibc;
+
+import "gogoproto/gogo.proto";
+import "dymension/common/common.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/dymensionxyz/dymension/v3/x/eibc/types";
+
+message DemandOrder {
+    // id is a hash of the form generated by GetRollappPacketKey, 
+    // e.g status/rollappid/packetProofHeight/packetDestinationChannel-PacketSequence which gurantees uniqueness
+    string id = 1;
```

Comment:
the id isn't a sequential numbering of the demandOrders?

Reply:
comes to solve the problem of:
1. being able to get demand order just by id
2. being able to fetch demand order by the undelrying rollapp packet it's associated with for update purposes in O(1)
3. not saving multiple keys for the same object (i.e one for the id (if it's a number) and one for the tracking packet)